packages/components/src/components/hds/dialog-primitive/footer.ts

```diff
@@ -15,14 +15,26 @@ export interface HdsDialogPrimitiveFooterSignature {
     default: [
       {
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        close?: (event: MouseEvent, ...args: any[]) => void;
+        close: (event: MouseEvent, ...args: any[]) => void;
       },
     ];
   };
   Element: HTMLDivElement;
 }
 
-const HdsDialogPrimitiveFooter =
-  templateOnlyComponent<HdsDialogPrimitiveFooterSignature>();
+const NOOP = (): void => {};
 
-export default HdsDialogPrimitiveFooter;
+export default class HdsDialogPrimitiveFooter extends Component<HdsDialogPrimitiveFooterSignature> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  get onDismiss(): (event: MouseEvent, ...args: any[]) => void {
+    const { onDismiss } = this.args;
+
+    // notice: this is to make sure the function is always defined when consumers add `{{on 'click' F.close}}` to a button in the DialogFooter.
+    // in reality it's always used inside the main components as a yielded component, so the onDismiss handler is always defined
+    if (typeof onDismiss === 'function') {
+      return onDismiss;
+    } else {
+      return NOOP;
```

Comment:
[question] I see that in the `Hds::Alert` component we have a very similar argument, but we use a slightly different approach (we simply return `false` in the getter, without the need to define a `noop` operation): https://github.com/hashicorp/design-system/blob/main/packages/components/src/components/hds/alert/index.ts 

Which one is more "correct™" (if there is one)?
Should we align this approach across the HDS components?

Reply:
Interesting... I got this implementation from [the DialogHeader](https://github.com/hashicorp/design-system/blob/36a45f0be2f6ea595e372e27e1d34a23774d19f9/packages/components/src/components/hds/dialog-primitive/header.ts#L40).

Looking at where the type errors are happening, if it returned false it wouldn't resolve the type issues.

Reply:
the `onDismiss` function in Alert is slightly different in a sense that it is not forwarded as `F.close` (or similar) so proper checks for it to actually exists and be a function are not required

Reply:
Makes sense, I've resolved the comment then